src/scripts/add-edit-proxy.js

```diff
@@ -1,198 +1,186 @@
+    const username = document.querySelector('#newProxyUsername').value.trim(); 
+    const password = document.querySelector('#newProxyPassword').value.trim();
+    if (username) { proxySetting.username = username; } // don't store ''
+    if (password) { proxySetting.password = password; } // don't store ''
   }
 
   // Set the password field type to text (not password) so that Firefox doesn't prompt
   // the user to save the password. Since we've already hidden this content with spinner,
   // user won't see the password anyway
-  var pwInput = $("#newProxyPassword");
-  $("#newProxyPassword").attr("type", "text");
+  document.querySelector('#newProxyPassword').setAttribute('type', 'text');
 
   if (oldProxySetting) {
     // Edit operation
     proxySetting.active = oldProxySetting.active;
     proxySetting.whitePatterns = oldProxySetting.whitePatterns;
     proxySetting.blackPatterns = oldProxySetting.blackPatterns;
     if (oldProxySetting.pacURL) proxySetting.pacURL = oldProxySetting.pacURL; // imported foxyproxy.xml
-    return editProxySetting(oldProxySetting.id, oldProxySetting.index, proxySetting)
+    return editProxySetting(oldProxySetting.id, oldProxySetting.index, proxySetting);
   }
   else {
     // Add operation
     proxySetting.active = true;  // new entries are instantly active. TODO: add checkbox on GUI instead of assuming
     // Do not use this proxy for internal IP addresses. TODO: add checkbox on GUI instead of assuming
-    if ($("#onOffWhiteAll").prop("checked")) proxySetting.whitePatterns = [PATTERN_ALL_WHITE];
-    else proxySetting.whitePatterns = [];
+    if (document.querySelector('#onOffWhiteAll').checked) { proxySetting.whitePatterns = [PATTERN_ALL_WHITE]; }
+    else { proxySetting.whitePatterns = []; }
 
-    if ($("#onOffBlackAll").prop("checked")) proxySetting.blackPatterns = [PATTERN_LOCALHOSTURLS_BLACK, PATTERN_INTERNALIPS_BLACK, PATTERN_LOCALHOSTNAMES_BLACK];
-    else proxySetting.blackPatterns = [];
+    if (document.querySelector('#onOffBlackAll').checked) {
+      proxySetting.blackPatterns = [PATTERN_LOCALHOSTURLS_BLACK, PATTERN_INTERNALIPS_BLACK, PATTERN_LOCALHOSTNAMES_BLACK];
+    }
+    else { proxySetting.blackPatterns = []; }
 
     return addProxySetting(proxySetting);
   }
 }
 
 function validateInput() {
   Utils.trimAllInputs();
-  Utils.escapeAllInputs("#newProxyTitle,#newProxyAddress,#newProxyPort");
-  let type = parseInt($("#newProxyType").val());
-  if (type == PROXY_TYPE_NONE) return true;
-  let r1 = markInputErrors("#newProxyAddress"), r2 = markInputErrors("#newProxyPort", true);
-  return r1 && r2;
+  Utils.escapeAllInputs('#newProxyTitle,#newProxyAddress,#newProxyPort');
+  const type = parseInt(document.querySelector('#newProxyType').value);
+  if (type === PROXY_TYPE_NONE) { return true; }
+  return markInputErrors('#newProxyAddress') && markInputErrors('#newProxyPort', true);
```

Comment:
You cannot return like this otherwise empty/invalid port will not be highlighted. I will revert to the previous code:
`let r1 = markInputErrors("#newProxyAddress"), r2 = markInputErrors("#newProxyPort", true);
return r1 && r2;`